enclave/src/top_pool/listener.rs

```diff
@@ -16,36 +16,50 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+use crate::{ocall::ocall_api::EnclaveRpcOCallApi, top_pool::watcher::Watcher};
+use codec::Encode;
 use linked_hash_map::LinkedHashMap;
 use log::{debug, trace};
 use sp_runtime::traits;
-use std::{collections::HashMap, fmt::Debug, hash, string::String, vec::Vec};
-
-use crate::top_pool::watcher::Watcher;
-use codec::Encode;
-
+use std::{collections::HashMap, fmt::Debug, hash, string::String, sync::Arc, vec::Vec};
 use substratee_worker_primitives::BlockHash as SidechainBlockHash;
 
 /// Extrinsic pool default listener.
-pub struct Listener<H: hash::Hash + Eq> {
-	watchers: HashMap<H, Watcher<H>>,
+pub struct Listener<H, R>
+where
+	H: hash::Hash + Eq,
+{
+	watchers: HashMap<H, Watcher<H, R>>,
 	finality_watchers: LinkedHashMap<SidechainBlockHash, Vec<H>>,
+	rpc_ocall_api: Arc<R>,
 }
 
 /// Maximum number of blocks awaiting finality at any time.
 const MAX_FINALITY_WATCHERS: usize = 512;
 
-impl<H: hash::Hash + Eq + Debug> Default for Listener<H> {
+impl<H, R> Default for Listener<H, R>
+where
+	H: hash::Hash + Eq + Debug,
+	R: EnclaveRpcOCallApi,
+{
 	fn default() -> Self {
```

Comment:
This simple default, we could most likely derive.

Reply:
yes indeed 👍